master/internal/experiment.go

```diff
@@ -484,25 +562,27 @@ func (e *experiment) isBestValidation(msg searcher.CompletedMessage) bool {
 	return isBest
 }
 
-func (e *experiment) updateState(ctx *actor.Context, state model.State) {
-	if wasPatched, err := e.Transition(state); err != nil {
+func (e *experiment) updateState(ctx *actor.Context, state model.State) (err error) {
+	var wasPatched bool
+	if wasPatched, err = e.Transition(state); err != nil {
 		ctx.Log().Error(err)
-		return
+		return err
 	} else if !wasPatched {
-		return
+		return nil
 	}
 	telemetry.ReportExperimentStateChanged(ctx.Self().System(), e.db, *e.Experiment)
 
 	ctx.Log().Infof("experiment state changed to %s", state)
 	for _, child := range ctx.Children() {
 		ctx.Tell(child, state)
 	}
-	if err := e.db.SaveExperimentState(e.Experiment); err != nil {
+	if err = e.db.SaveExperimentState(e.Experiment); err != nil {
```

Comment:
blocking: in experiment.go, we ignore most database errors and continue forward with the experiment. Take a look a some of the other database calls in experiment.go (other than in PostStop which is a special case).

Reply:
yeah I was surprised by that, why is that?